src/static/js/AttributeManager.js

```diff
@@ -102,9 +102,11 @@ AttributeManager.prototype = _(AttributeManager.prototype).extend({
         this.rep.lines.offsetOfIndex(row + 1) - this.rep.lines.offsetOfIndex(row) - 1;
     const markerWidth = this.lineHasMarker(row) ? 1 : 0;
     if (lineLength - markerWidth < 0) throw new Error(`line ${row} has negative length`);
-
     const startCol = row === start[0] ? start[1] : markerWidth;
-    if (startCol - markerWidth < 0) throw new RangeError('selection starts before line start');
+
+    if ((startCol !== 0) && (startCol - markerWidth < 0)) {
+      throw new RangeError('selection starts before line start');
+    }
```

Comment:
It absolutely is, its a hammer to fix a vase.

Reply:
I think it would be better to adjust the column past the line marker so that the behavior is consistent across lines:

```javascript
    if (start[1] < 0) throw new RangeError('selection starts at negative column');
    const startCol = Math.max(markerWidth, row === start[0] ? start[1] : 0);
    if (startCol > lineLength) throw new RangeError('selection starts after line end');

    if (end[1] < 0) throw new RangeError('selection ends at negative column');
    const endCol = Math.max(markerWidth, row === end[0] ? end[1] : lineLength);
    if (endCol > lineLength) throw new RangeError('selection ends after line end');
    if (startCol > endCol) throw new RangeError('selection ends before it starts');
```